fsspec/asyn.py

```diff
@@ -360,15 +360,16 @@ async def _cat(self, path, recursive=False, on_error="raise", **kwargs):
 
     async def _cat_ranges(self, paths, starts, ends, max_gap=None, **kwargs):
         # TODO: on_error
-        if max_gap is not None:
-            # to be implemented in utils
-            raise NotImplementedError
         if not isinstance(paths, list):
             raise TypeError
         if not isinstance(starts, list):
             starts = [starts] * len(paths)
         if not isinstance(ends, list):
             ends = [starts] * len(paths)
+        if max_gap is not None:
+            paths, starts, ends = merge_offset_ranges(
```

Comment:
The problem here is, that the returned list does not align with the input list; we need a way to tell the caller what ranges we are actually fetching. *Unless* we run a second pass and unbundle the bytes blocks into the original start/stop lists (this would involve memory copies, at least, and the pieces might get stuck back together again if used as the cache for a file).

Reply:
Ah - Great point.  The user is asking to merge ranges, but still needs to know what those ranges are after merging occurs.  In the interest of keeping the `cat_ranges` API simple, perhaps the best solution is to remove the `max_gap=` argument, and require the user to call `merge_offset_ranges` seperately?

Reply:
That's definitely the easiest option

Reply:
>the code would already be useful, but require the called to themselves call marge_offset_ranged, and then cat_ranges with the result (i.e., not actually use the max_gap arg of the latter).

I guess you already pointed this out :)